kernel/syscall.c

```diff
@@ -2815,20 +2815,32 @@ long myst_syscall_prlimit64(
     if (pid)
         return -EINVAL;
 
-    // Only support resource NOFILE
-    if (resource != RLIMIT_NOFILE)
-        return -EINVAL;
+    if (resource == RLIMIT_NOFILE)
+    {
+        if (old_rlim)
+        {
+            // ATTN: return currently effective RLIMIT_NOFILE settings
+            old_rlim->rlim_cur = 65536;
+            old_rlim->rlim_max = 65536;
+        }
 
-    if (old_rlim)
+        if (new_rlim)
+        {
+            // ATTN: make RLIMIT_NOFILE settings effective ;
+        }
+    }
+    else if (resource == RLIMIT_STACK)
     {
-        // ATTN: return currently effective RLIMIT_NOFILE settings
-        old_rlim->rlim_cur = 65536;
-        old_rlim->rlim_max = 65536;
+        if (old_rlim)
+        {
+            // TODO: get actual stack limits
```

Comment:
Mystikos process' main thread is initialized with a default stack size in [kernel/exec.c](https://github.com/deislabs/mystikos/blob/87c7a2ec370115ac35452fa1dd3fffe0c544d044/kernel/exec.c#L838). Probably should define a DEFAULT_PROCESS_STACK_SIZE and use here and in kernel/exec.c. 

@mikbras , any suggestion on which header file to include the DEFAULT_PROCESS_STACK_SIZE definition?

Reply:
```suggestion
```